hbase-it/src/test/java/org/apache/hadoop/hbase/backup/IntegrationTestBackupRestoreBase.java

```diff
@@ -0,0 +1,748 @@
+        }
+        LOG.info("Waiting up to {} seconds for WAL partition directory to exist: {}", waitTimeSec,
+          walPartitionDir);
+        Thread.sleep(1000);
+      }
+    }
+    return walPartitionDir;
+  }
+
+  /**
+   * Verifies the WAL partition directory contains a backup WAL file. The WAL file's path will look
+   * something like the following:
+   * .../backupWALDir/WALs/2025-10-17/wal_file.1760738249595.1880be89-0b69-4bad-8d0e-acbf25c63b7e
+   * @param walPartitionDir The date directory for a backup WAL i.e.
+   *                        .../backupWALDir/WALs/2025-10-17
+   */
+  private void verifyBackupWALFiles(Path walPartitionDir) throws IOException {
+    FileStatus[] fileStatuses = fs.listStatus(walPartitionDir);
+    for (FileStatus fileStatus : fileStatuses) {
+      String walFileName = fileStatus.getPath().getName();
+      String[] splitName = walFileName.split("\\.");
+      assertEquals("The WAL partition directory should only have files that start with 'wal_file'",
+        "wal_file", splitName[0]);
+      assertEquals(
+        "The timestamp in the WAL file's name should match the date for the WAL partition directory",
+        walPartitionDir.getName(), BackupUtils.formatToDateString(Long.parseLong(splitName[1])));
+    }
+  }
+
+  /**
+   * Checks if all timestamps in a map with ServerName and timestamp key-values pairs are past the
+   * provided timestamp threshold.
+   * @param timestamps Map with ServerName and timestamp key-value pairs
+   * @param threshold  Timestamp to check all timestamp values in the map against
+   * @return True if all timestamps values in the map have met or are path the timestamp threshold;
+   *         False otherwise
+   */
+  private boolean areAllTimestampsPastThreshold(Map<ServerName, Long> timestamps, long threshold,
+    TableName tableName) {
+    boolean haveAllTimestampsReachedThreshold = true;
+    LOG.info(
+      "Checking if each region server in the replication check point has caught up to the latest Put on {}",
+      tableName.getNameAsString());
+    for (Map.Entry<ServerName, Long> entry : timestamps.entrySet()) {
+      LOG.info("host={}, checkpoint timestamp={}, latest put timestamp={}, caught up={}",
+        entry.getKey(), entry.getValue(), threshold, entry.getValue() >= threshold);
+      if (entry.getValue() < threshold) {
+        // Not returning right away so all hosts and timestamps are logged
+        haveAllTimestampsReachedThreshold = false;
+      }
+    }
+    return haveAllTimestampsReachedThreshold;
+  }
+
+  /**
+   * Waits for the replication checkpoint timestamp of each region server to meet or pass the
+   * timestamp of the latest Put operation on the backed-up table.
+   * @param conn               Minicluster connection
+   * @param latestPutTimestamp Timestamp of the latest Put operation on the backed-up table
+   */
+  private void waitForCheckpointTimestampUpdate(Connection conn, long latestPutTimestamp,
```

Comment:
This method is quite misleading. You cannot wait for multiple region servers by the latest timestamp. This only works with strictly 1 region server.

If you have multiple region servers, the data that you loaded is distributed across the RSs. Once data is replicated by all of them the timestamp marker will be the timestamp of latest entry exactly on one RS. All others have timestamps which are smaller than this. In other words the data you've loaded is split into multiple groups (3, 5, 6), (1, 2, 4), (7, 8, 9), ... Timestamp markers will be [6, 4, 9] when replication is finished. You cannot wait for timestamp (9) on all region servers.

Reply:
The test runs for me endlessly.

Reply:
Worked for me too in CLI:
```
 Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1315 s -- in org.apache.hadoop.hbase.backup.IntegrationTestContinuousBackupRestore
```